src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/StorePal.cs

```diff
@@ -17,7 +17,7 @@ namespace Internal.Cryptography.Pal
 {
     internal sealed partial class StorePal
     {
-        private static CollectionBackedStoreProvider s_machineRootStore;
+        private static volatile CollectionBackedStoreProvider s_machineRootStore;
```

Comment:
Why is this necessary?

Reply:
Thanks, but that doesn't answer my question. What reordering does this prevent that would otherwise cause a bug?